### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by corleonechensiyu**
lstm测试有效，参考https://github.com/kasyoukin/caffe_ocr_for_linux/blob/master/src/caffe/layers/lstm_layer_Junhyuk.cpp

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=1527) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=1527) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=1527) it.</sub>

**Comment by corleonechensiyu** [comment]
ocr需要的lstm，reverse及innerproduct，axis为2的操作

**Comment by corleonechensiyu** [comment]
> lstm可以直接改，不用新加lstmnew
> 新文件开头要加copyright，可以参考其他文件的写法
> 不要用tab，缩进保持4个空格
> src/layer/arm/innerproduct_arm.cpp有好多注释代码，如果没有用请删掉
> 新加的层在src/CMakeLists.txt 中增加

嗯嗯，好的，第一次pull requests ：）

File: src/layer/arm/innerproduct_arm.cpp [hunk 1]
```diff
@@ -353,7 +354,118 @@ int InnerProduct_arm::forward(const Mat& bottom_blob, Mat& top_blob, const Optio
 
         top_blob[p] = sum;
     }
-
+   }
+    else if(axis==2)
+    {
```
Comment (nihui) [inline src/layer/arm/innerproduct_arm.cpp]: There is not an implementation for packing layout

You have to claim this in create_pipeline()
```
if (axis == 1)
    support_packing = false;
```

**Comment by nihui** [comment]
onnx lstm

**Pull request closed without merging**